client/landing/stepper/declarative-flow/flows/new-hosted-site-flow/new-hosted-site-flow.ts

```diff
@@ -14,34 +14,70 @@ import {
 	setSignupCompleteSiteID,
 	getSignupCompleteSlug,
 } from 'calypso/signup/storageUtils';
-import { useDispatch as reduxUseDispatch, useSelector } from 'calypso/state';
 import { isUserEligibleForFreeHostingTrial } from 'calypso/state/selectors/is-user-eligible-for-free-hosting-trial';
 import { setSelectedSiteId } from 'calypso/state/ui/actions';
 import { useQuery } from '../../../hooks/use-query';
 import { ONBOARD_STORE } from '../../../stores';
 import { stepsWithRequiredLogin } from '../../../utils/steps-with-required-login';
 import { STEPS } from '../../internals/steps';
-import { Flow, ProvidedDependencies } from '../../internals/types';
-import type { OnboardSelect } from '@automattic/data-stores';
+import type { FlowV2, ProvidedDependencies, StepperStep } from '../../internals/types';
+import type { OnboardActions, OnboardSelect } from '@automattic/data-stores';
 
-function useShowDomainStep(): boolean {
-	const query = useQuery();
-	return query.has( 'showDomainStep' );
-}
-
-const hosting: Flow = {
+const hosting: FlowV2 = {
 	name: NEW_HOSTED_SITE_FLOW,
 	__experimentalUseBuiltinAuth: true,
 	isSignupFlow: true,
-	useSteps() {
-		const showDomainStep = useShowDomainStep();
-		return stepsWithRequiredLogin( [
-			...( showDomainStep ? [ STEPS.UNIFIED_DOMAINS ] : [] ),
-			STEPS.UNIFIED_PLANS,
-			STEPS.TRIAL_ACKNOWLEDGE,
-			STEPS.SITE_CREATION_STEP,
-			STEPS.PROCESSING,
-		] );
+	async initialize( reduxStore ) {
+		const { resetOnboardStore, setPlanCartItem } = dispatch( ONBOARD_STORE ) as OnboardActions;
+
+		await resetOnboardStore();
+
+		// @ts-expect-error - This is a thunk. We need to type `reduxStore` better.
+		reduxStore.dispatch( setSelectedSiteId( null ) );
```

Comment:
My suggestion is also bad but I think it's less bad than ignoring the error.

```suggestion
		reduxStore.dispatch( setSelectedSiteId( null ) as unknown as AnyAction );
```

Although I'm not sure we need this at all. Stepper seems to[ reset the site ID on boot](https://github.com/Automattic/wp-calypso/pull/102659/files#diff-7b49b33bf8ea9f6d6f6633e847a14d158dfab4c9ef7578508dcbda343b9a9437R133).